DepositRegistrar.sol

```diff
@@ -0,0 +1,306 @@
+        value = _value;
+        if (!bidder.send(this.balance - _value)) throw;
+    }
+    
+    function closeDeed( uint burnAmount) noEther onlyRegistrar {
+        burn.send((burnAmount * this.balance)/1000);
+        selfdestruct(bidder);
+    }
+    
+    function setBalance(uint newValue) onlyRegistrar {
+        // Check if it has enough balance to set the value
+        if (this.balance < newValue) throw;
+        // Send the difference to the owner
+        if (!owner.send(this.balance - newValue)) throw;
+    }
+    
+    function transferOwnership(address newOwner) {
+        if (msg.sender != owner ) throw;
+        owner = newOwner;
+    }
+}
+
+contract Registrar {
+    mapping (bytes32 => entry) public hashedNames;
+    mapping (bytes32 => address) public sealedBids;
+    
+    enum Mode { Open, Auction, Owned }
+    uint auctionLength = 20 minutes;
+    uint revealPeriod = 5 minutes;
+    uint renewalPeriod = 2 hours;
+    uint M = 1000000; // just a multiplier to get more precision on averages
+    uint public averagePeriod;
+    uint public lastSinceNewName;
+    uint public minBid = 0.1 ether;    // TODO: figure out a better way to come up with this
+    uint public averagePrice = minBid;
+    
+    struct entry {
+        Mode status;
+        address deed;
+        uint auctionDate;
+        uint value;
+        uint highestBid;
+        uint firstRegistered;
+        uint lastRenewed;
+        uint renewalDate;
+        uint averagePrice;
+    }
+    
+    // doesnt work
+    modifier withName(bytes32 nameHash){
+        entry name = hashedNames[nameHash];
+        _        
+    }
+    
+    modifier noEther  {
+        if (msg.value > 0) throw;
+        _
+    }   
+    
+    function Registrar() noEther  {
+        lastSinceNewName = now;
+    }
+
+    /*
+    ## Start Auction for available name
+
+    Anyone can start an auction by sending an array of hashes that they want to bid for. 
+    Arrays are sent so that someone can open up an auction for X dummy hashes when they 
+    are only really interested in bidding for one. This will increase the cost for an 
+    attacker from simply bidding on all new auctions blindly. Dummy auctions that are 
+    open but not bid on are closed after a week. 
+    */    
+    function openAuctions(bytes32[] auctionsArray) noEther{
+
+        for (uint i = 0; i < auctionsArray.length; i ++ ) {
+            entry newAuction = hashedNames[auctionsArray[i]];
+            if (newAuction.status == Mode.Owned && now < newAuction.renewalDate) {
+                throw; 
+            } else if (newAuction.status == Mode.Auction && now < newAuction.auctionDate) {
+                throw;
+            } else if (newAuction.status == Mode.Owned) {
+                Deed deedContract = Deed(newAuction.deed);
+                deedContract.closeDeed(1);
+            }
+            newAuction.auctionDate = now + auctionLength;
+            newAuction.renewalDate = now + renewalPeriod;
```

Comment:
This was necessary before, but now after some rewrite I don't see why not remove it